### PR TITLE
Add -bytecode-compiler flag for enabling vm_compute in coqchk

### PR DESCRIPTION
**Pull request by Alizter**
I also cleaned up the --help for coqchk

- [x] Added **changelog**.
- [x] Added / updated **documentation**.

### CODE REVIEW

File: checker/checkFlags.ml [hunk 1]
```diff
@@ -11,13 +11,19 @@
...
 let set_local_flags flags env =
+  (* Explicitly ignored flags have been commented out *)
   let flags =
     { (Environ.typing_flags env) with
       check_guarded = flags.check_guarded;
       check_positive = flags.check_positive;
       check_universes = flags.check_universes;
       conv_oracle = flags.conv_oracle;
       share_reduction = flags.share_reduction;
+      enable_VM = flags.enable_VM;
```
Comment (SkySkimmer) [inline checker/checkFlags.ml:22]: This will enable the vm regardless of the command line flag

File: checker/checkFlags.ml [hunk 2]
```diff
@@ -11,13 +11,19 @@
 open Declarations
 
 let set_local_flags flags env =
+  (* Explicitly ignored flags have been commented out *)
```
Comment (SkySkimmer) [inline checker/checkFlags.ml:14]: until we add some in the datatype

File: checker/checkFlags.ml (hunk 1 above)
Comment (Alizter) [reply checker/checkFlags.ml:22]: Reading the code, I was under the impression that make_senv was in charge of setting flags. This is to override specific flags no?

File: checker/checkFlags.ml (hunk 2 above)
Comment (Alizter) [reply checker/checkFlags.ml:14]: Right, so we should perhaps do this a better way.

Comment (Alizter) [reply checker/checkFlags.ml]: Done.

File: checker/checkFlags.ml (hunk 1 above)
Comment (Alizter) [reply checker/checkFlags.ml]: Is enable_VM not set to true by default (in coqc)? Wouldn't that mean it was always enabled?

File: checker/checkFlags.ml [hunk 3]
```diff
@@ -11,15 +11,21 @@
...
+    check_universes = flags.check_universes;
+    conv_oracle = flags.conv_oracle;
+    share_reduction = flags.share_reduction;
+    enable_VM = flags.enable_VM;
+    enable_native_compiler = envflags.enable_native_compiler;
+    indices_matter = envflags.indices_matter;
+    impredicative_set = envflags.impredicative_set;
+    sprop_allowed = envflags.sprop_allowed;
+    cumulative_sprop = flags.cumulative_sprop;
+    allow_uip = flags.allow_uip;
```
Comment (SkySkimmer) [inline checker/checkFlags.ml:28]:
```suggestion
    check_guarded = flags.check_guarded;
    check_positive = flags.check_positive;
    check_universes = flags.check_universes;
    conv_oracle = flags.conv_oracle;
    share_reduction = flags.share_reduction;
    cumulative_sprop = flags.cumulative_sprop;
    allow_uip = flags.allow_uip;
    
    enable_VM = envflags.enable_VM && flags.enable_VM;
    
    enable_native_compiler = envflags.enable_native_compiler;
    indices_matter = envflags.indices_matter;
    impredicative_set = envflags.impredicative_set;
    sprop_allowed = envflags.sprop_allowed;
```

File: checker/checkFlags.ml (hunk 1 above)
Comment (Alizter) [reply checker/checkFlags.ml]: Do we really want to be able to override this here anyway?

Comment (Alizter) [reply checker/checkFlags.ml]: Shouldn't this just be `envflags.enable_VM`?

**Comment by Alizter** [comment]
@SkySkimmer Can you assign?

Do you think it needs a changelog?

**Comment by SkySkimmer** [comment]
>Do you think it needs a changelog?

yes

File: checker/checker.ml [hunk 4]
```diff
@@ -178,23 +180,27 @@ let print_usage_channel co command =
...
+\n  -v, --version               print coqchk version and exit\
+\n  -o, --output-context        print the list of assumptions\
+\n  -m, --memory                print the maximum heap size\
+\n  -silent                     disable trace of constants being checked\
 \n\
-\n  -impredicative-set     set sort Set impredicative\
+\n  -impredicative-set          set sort Set impredicative\
+\n  -indices-matter             levels of indices (and nonuniform parameters)\
+\n                              contribute to the level of inductives\
+\n  -bytecode-compiler (yes|no) enable the vm_compute reduction machine\
```
[the remark this answers is not in the document]
Comment (Alizter) [reply checker/checker.ml:201]: coqc doesn't say it is on by default, should I change that too?

[the remark this answers is not in the document]
Comment (SkySkimmer) [reply checker/checker.ml:201]: you decide

[the remark this answers is not in the document]
Comment (Alizter) [reply checker/checker.ml:201]: I'm not confident enough to change the coqc one since I'm not really sure what it means there for the flag to be on.

**Review by SkySkimmer — APPROVED** [review]

**Pull request merged**